### ISSUE TITLE
Consider naming WRT literals to be consistent with RDF 1.1 and XSD 1.1 specs

### ISSUE THREAD

**Issue by wouterbeek** [rdfjs/representation-task-force#13]
Currently the spec defines the following three properties for literals:
- `.value` the text value, unescaped, without language or type
- `.language` the language as lowercase BCP47 string
- `.datatype` the datatype IRI as string

What is called a "value" here is called a "lexical expression" by the RDF 1.1 and XSD 1.1 specs.  Also, the latter uses "value" for the interpretation of a literal.  For instance, the literal `"01"^^xsd:int` has lexical expression `"01"`, datatype IRI `xsd:int`, no language tag, and value `1`.

**Comment by RubenVerborgh** [comment]
:+1:

**Comment by RubenVerborgh** [comment]
This also begs the question whether we want a separate construct for “value”, which would then be `1` (number) in your example. Could be very handy to have these as getters/setters.

**Comment by westurner** [comment]
* RDF 1.1 Literal: http://www.w3.org/TR/rdf11-concepts/#section-Graph-Literal

* RDF 1.1 datatype IRIs: http://www.w3.org/TR/rdf11-concepts/#datatype-iris
  * RDF 1.1 XSD 1.1 datatypes: http://www.w3.org/TR/rdf11-concepts/#xsd-datatypes
    *   "W3C XML Schema Definition Language (XSD) 1.1 Part 2: Datatypes" http://www.w3.org/TR/xmlschema11-2/
  * ``rdf:HTML`` datatype IRI: http://www.w3.org/TR/rdf11-concepts/#section-html
  * ``rdf:XMLLiteral`` datatype IRI: http://www.w3.org/TR/rdf11-concepts/#section-XMLLiteral

**Comment by westurner** [comment]
{``rdfx:Markdown``, ``rdfx:ReStructuredText``, ``rdfx:MediaWiki``} with a ``.html`` ``rdf:HTML`` could be convenient.

**Comment by westurner** [comment]
* [[ ]] ``.unit`` IRIs from e.g. QUDT would also be great for RDF 1.X

**Comment by RubenVerborgh** [comment]
See #17 for the  separate construct for a (native) value.

**Comment by ktk** [comment]
+1 for adding datatypes like Markdown, very useful in a webby environment.

**Comment by westurner** [comment]
On Dec 15, 2015 4:06 AM, "Adrian Gschwend" <notifications@github.com> wrote:
>
> +1 for adding datatypes like Markdown, very useful in a webby environment.

>> [[ ]] {rdfx:Markdown, rdfx:ReStructuredText, rdfx:MediaWiki} with a
.html rdf:HTML could be convenient.

In which RDFS vocabularies do you think  these :markdown, :restructuredtext
classes and properties would be most useful?

**Comment by ericprud** [comment]
* Wes Turner <notifications@github.com> [2015-12-15 02:12-0800]
> On Dec 15, 2015 4:06 AM, "Adrian Gschwend" <notifications@github.com> wrote:
> >
> > +1 for adding datatypes like Markdown, very useful in a webby environment.
> 
> >> [[ ]] {rdfx:Markdown, rdfx:ReStructuredText, rdfx:MediaWiki} with a
> .html rdf:HTML could be convenient.
> 
> In which RDFS vocabularies do you think  these :markdown, :restructuredtext
> classes and properties would be most useful?

RDF. It sets the precedent with rdf:XMLLiteral.

The partitioning of rdf: and rdfs: has been a bit of an obstacle The
RDF served at <http://www.w3.org/2000/01/rdf-schem> define these
terms:

Subsumption            : rdfs:subClassOf, rdfs:subPropertyOf, rdfs:domain, rdfs:range
Containers             : rdfs:Container, rdfs:ContainerMembershipProperty, rdfs:member, rdfs:Datatype
Structural description : rdfs:Literal, rdfs:Resource, rdfs:Class
Annotation             : rdfs:comment, rdfs:label, rdfs:seeAlso, rdfs:isDefinedBy


IMO, the only terms that should be in the RDFS namespace are the
subsumption properties, possibly including container subsumption.


> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/rdfjs/representation-task-force/issues/13#issuecomment-164712514

-- 
-ericP

office: +1.617.599.3509
mobile: +33.6.80.80.35.59

(eric@w3.org)
Feel free to forward this message to any list for any purpose other than
email address distribution.

There are subtle nuances encoded in font variation and clever layout
which can only be seen by printing this message on high-clay paper.

**Comment by westurner** [comment]
On Dec 15, 2015 5:13 AM, "ericprud" <notifications@github.com> wrote:
>
> * Wes Turner <notifications@github.com> [2015-12-15 02:12-0800]
>
> > On Dec 15, 2015 4:06 AM, "Adrian Gschwend" <notifications@github.com>
wrote:
> > >
> > > +1 for adding datatypes like Markdown, very useful in a webby
environment.
> >
> > >> [[ ]] {rdfx:Markdown, rdfx:ReStructuredText, rdfx:MediaWiki} with a
> > .html rdf:HTML could be convenient.
> >
> > In which RDFS vocabularies do you think these :markdown,
:restructuredtext
> > classes and properties would be most useful?
>
> RDF. It sets the precedent with rdf:XMLLiteral.

* What's the refresh cycle of RDF?
* When would e.g an :LightweightMarkupLanguage > :ReStructuredText reach
downstream tooling?
* Are there good ways for provisionally defining said properties which
would make it easier to reach final draft?
  * ``rdfx-local:``

**Comment by wouterbeek** [comment]
The datatype IRI for Markdown already exists.  It's [`sysont:Markdown`](http://ns.ontowiki.net/SysOnt/Markdown).  The problem with it is that its semantic description is very poor (and "formated" should probably be "formatted", but that is merely a syntactic mistake):

```turtle
sysont:Markdown a rdfs:Datatype ;
  rdfs:comment "A string literal formated using markdown syntax." ;
  rdfs:label   "Markdown formated string" ;
  rdfs:seeAlso "http://daringfireball.net/ projects/markdown/syntax" .
```

The XSD datatypes, as well as `rdf:XMLLiteral` and `rdf:HTML`, have a formal semantics.  For instance for `rdf:HTML` "The value space  is a set of DOM DocumentFragment nodes [DOM4]. Two DocumentFragment nodes A and B are considered equal if and only if the DOM method A.isEqualNode(B) [DOM4] returns true."  One may argue whether this natural language variant of imperative programming is the best way of defining semantics, as discussed in [http://homepages.cwi.nl/~steven/Talks/2015/11-06-xml-amsterdam/]("HTML5 is the new Flash"), but it is at least something that can be implemented for automated processing.

If I'm correct there is no formal grammar or semantics for Markdown, so it may not even be possible to semantically specify the meaning of this datatype IRI.  It may of course mean something like "This string is formatted in a way that is commonly termed 'Markdown'.  It may be processed by some/many tools that claim to processes 'Markdown'."  The big problem with this 'definition' is that a machine processor is unable to automatically determine, e.g., whether a 'Markdown' string found on the Semantic Web can indeed be parsed by a 'Markdown' tool.  The XSD and RDF datatypes are guaranteed to be fully machine processable.

**Comment by westurner** [comment]
On Dec 15, 2015 5:52 AM, "Wouter Beek" <notifications@github.com> wrote:
>
> The datatype IRI for Markdown already exists. It's sysont:Markdown. The
problem with it is that its semantic description is very poor (and
"formated" should probably be "formatted", but that is merely a syntactic
mistake):

All I want to reach for is `schema.org`.

>
> sysont:Markdown a rdfs:Datatype ;
>   rdfs:comment "A string literal formated using markdown syntax." ;
>   rdfs:label   "Markdown formated string" ;
>   rdfs:seeAlso "http://daringfireball.net/ projects/markdown/syntax" .
>
> The XSD datatypes, as well as rdf:XMLLiteral and rdf:HTML, have a formal
semantics. For instance for rdf:HTML "The value space is a set of DOM
DocumentFragment nodes [DOM4]. Two DocumentFragment nodes A and B are
considered equal if and only if the DOM method A.isEqualNode(B) [DOM4]
returns true." One may argue whether this natural language variant of
imperative programming is the best way of defining semantics, as discussed
in http://homepages.cwi.nl/~steven/Talks/2015/11-06-xml-amsterdam/, but it
is at least something that can be implemented for automated processing.
>

```
:markdown_github
:markdown_commonmark
Markdown > MarkdownGithub
Markdown > MarkdownCommonmark
```

```
:markdown_commonmark_html a
  rdfs:range[ rdf:HTML , rdf:XMLLiteral , ];
```

> If I'm correct there is no formal grammar or semantics for Markdown, so
it may not even be possible to semantically specify the meaning of this
datatype IRI. It may of course mean something like "This string is
formatted in a way that is commonly termed 'Markdown'. It may be processed
by some/many tools that claim to processes 'Markdown'." The big problem
with this 'definition' is that a machine processor is unable to
automatically determine, e.g., whether a 'Markdown' string found on the
Semantic Web can indeed be parsed by a 'Markdown' tool. The XSD and RDF
datatypes are guaranteed to be fully machine processable.

From
* https://westurner.org/tools/#markdown
* https://westurner.org/tools/#commonmark
::

Markdown
++++++++++
| Wikipedia: https://en.wikipedia.org/wiki/Markdown
| Homepage: https://daringfireball.net/projects/markdown/
| Standard: https://daringfireball.net/projects/markdown/syntax
| Docs: http://www.w3.org/community/markdown/wiki/MarkdownImplementations
| Docs: https://en.wikipedia.org/wiki/Markdown#Implementations
| Docs: http://learnxinyminutes.com/docs/markdown/
| Docs: https://guides.github.com/features/mastering-markdown/
| Docs: https://help.github.com/articles/github-flavored-markdown/
| Docs: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet
| FileExt: ``.Md``
| FileExt: ``.md``

Markdown is a :ref:`Lightweight markup language`
which can be parsed and transformed to
valid :ref:`HTML-`.

* GitHub and BitBucket support Markdown
  in Issue Descriptions, Wiki Pages, and Comments
* :ref:`Jupyter Notebook` supports Markdown
  in Markdown cells

.. index:: CommonMark
.. _commonmark:

CommonMark
+++++++++++++
| Homepage: http://commonmark.org
| Standard: http://spec.commonmark.org/0.22/
| Source:  https://github.com/jgm/CommonMark

:ref:`CommonMark` is one effort to standardize :ref:`Markdown`.

**Comment by wouterbeek** [comment]
@westurner I don't understand your post.  Did something go wrong in your (Markdown) markup?

**Comment by bergos** [comment]
Closed based on the resolution in #136
